### PR TITLE
Upgrade to go1.22 and other updates

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary

- upgrade to go1.22
- add job to build with gorelease to test the flow
- group dependabot updates
- drop job that is not useful

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -10,11 +10,11 @@ env:
...
-    - /bin/bash -c 'if [ -n "$(git --no-pager diff --exit-code go.mod go.sum)" ]; then exit 1; fi'
-# if running a release we will generate the images in this step
-# if running in the CI the CI env va is set by github action runner and we dont run the ko steps
-# this is needed because we are generating files that goreleaser was not aware to push to GH project release
-    - /bin/bash -c 'if [ -n "$CI" ]; then make sign-container-release; fi'
+    # - /bin/bash -c 'if [ -n "$(git --no-pager diff --exit-code go.mod go.sum)" ]; then exit 1; fi'
+    # if running a release we will generate the images in this step
+    # if running in the CI the CI env va is set by github action runner and we dont run the ko steps
+    # this is needed because we are generating files that goreleaser was not aware to push to GH project release
+    # - /bin/bash -c 'if [ -n "$CI" ]; then make sign-container-release; fi'
```
Comment (haydentherapper) [inline .goreleaser.yml:17]: Just to double check, this is not needed?

Comment (cpanato) [reply .goreleaser.yml:17]: it is, my bad

Comment (cpanato) [reply .goreleaser.yml]: did a small refactor

**Review by haydentherapper — DISMISSED** [review]
Thank you! Have you tested this release on a fork to confirm the GHAs are working as expected?

**Comment by cpanato** [comment]
> Thank you! Have you tested this release on a fork to confirm the GHAs are working as expected?

will do that just to confirm :)

**Comment by cpanato** [comment]
here @haydentherapper 

job: https://github.com/cpanato/timestamp-authority/actions/runs/9059291147
image: ghcr.io/cpanato/timestamp-server:v99.99.00
release: https://github.com/cpanato/timestamp-authority/releases/tag/v99.99.00

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**